### PR TITLE
Correct failure of netty build component on s390x (due to zip -qd needing something to delete)

### PR DESCRIPTION
**Pull request by jonpspri**
Build on s390x fails because the process to remove unnecessary libraries from netty was not configured for x86.  As a result, zip -d failed because it had nothing to remove.  This PR does two things:

1.  Adds conditions for s390x to the src and third_party BUILD files, and
2.  Adds a condition to the genrule so that 'zip -d' does not execute if there's nothing to do.

See also Issue #9263; Pull Requests #9346 and #9945

### CODE REVIEW

File: third_party/BUILD [hunk 1]
```diff
@@ -469,9 +469,9 @@ UNNECESSARY_DYNAMIC_LIBRARIES = select({
     "//src/conditions:arm": "*.so *.jnilib *.dll",
     "//src/conditions:linux_aarch64": "*.so *.jnilib *.dll",
     "//src/conditions:linux_ppc": "*.so *.jnilib *.dll",
+    "//src/conditions:linux_s390x": "*.so *.jnilib *.dll",
     # Play it safe -- better have a big binary than a slow binary
-    # zip -d does require an argument. Supply something bogus.
-    "//conditions:default": "*.bogusextension",
+    "//conditions:default": "",
```
Comment (gregestren) [inline third_party/BUILD]: This fulfills point #2, right?

Comment (jonpspri) [reply third_party/BUILD]: Correct.  The empty string means nothing is removed.  The test for the empty string in the shell execution keeps unzip from being run as a no-op (causing a failure).

Comment (gregestren) [reply third_party/BUILD]: Thanks for clarifying!

**Comment by gregestren** [comment]
Jonathan,

It looks like we have to follow a different merge process for patches that include `third_party`. I can merge `src/conditions/BUILD` in this PR, then we'll need a separate PR just for `third_party/BUILD`.

Does that ordering sequence work and do you mind setting up the PR structure that way?

**Comment by jonpspri** [comment]
@gregestren Yes, that works.  Go ahead and merge `src/conditions/BUILD` from this PR.  Once it's in place and percolated through, I'll open a new PR for `third_party/BUILD`.  It looks like there's a conflict there to resolve anyway.

Do I need to undo the third_party/BUILD changes in this branch for that to work?

**Comment by gregestren** [comment]
I believe it's okay as-is. I'll try to merge - I'll add a comment here if anything doesn't work.

**Comment by gregestren** [comment]
This patch is in - ready for third_party/BUILD when you are.

**Comment by jonpspri** [comment]
@gregestren Perfect.  I'll rebase and submit a new pull request later today.  Kinda busy doing my "real job" this morning.

**Pull request closed without merging**